File: chain-sdk/clojure/src/on_chain_api/specs/token_v3_day_data_dto.clj
(ns on-chain-api.specs.token-v3-day-data-dto
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            )
  (:import (java.io File)))


(def token-v3-day-data-dto-data
  {
   (ds/opt :entry_time) inst?
   (ds/opt :recv_time) inst?
   (ds/opt :block_number) int?
   (ds/opt :vid) int?
   (ds/opt :id) string?
   (ds/opt :date) int?
   (ds/opt :token) string?
   (ds/opt :volume) string?
   (ds/opt :volume_usd) string?
   (ds/opt :untracked_volume_usd) string?
   (ds/opt :total_value_locked) string?
   (ds/opt :total_value_locked_usd) string?
   (ds/opt :price_usd) string?
   (ds/opt :fees_usd) string?
   (ds/opt :open) string?
   (ds/opt :high) string?
   (ds/opt :low) string?
   (ds/opt :close) string?
   })

(def token-v3-day-data-dto-spec
  (ds/spec
    {:name ::token-v3-day-data-dto
     :spec token-v3-day-data-dto-data}))

File: chain-sdk/javascript-closure-angular/API/Client/TokenV3DayDataDTO.js
goog.provide('API.Client.TokenV3DayDataDTO');

/**
 * Token data aggregated across all pairs that include token.
 * @record
 */
API.Client.TokenV3DayDataDTO = function() {}

/**
 * @type {!Date}
 * @export
 */
API.Client.TokenV3DayDataDTO.prototype.entryTime;

/**
 * @type {!Date}
 * @export
 */
API.Client.TokenV3DayDataDTO.prototype.recvTime;

/**
 * 
 * @type {!number}
 * @export
 */
API.Client.TokenV3DayDataDTO.prototype.blockNumber;

/**
 * 
 * @type {!number}
 * @export
 */
API.Client.TokenV3DayDataDTO.prototype.vid;

/**
 * Token address concatendated with date.
 * @type {!string}
 * @export
 */
API.Client.TokenV3DayDataDTO.prototype.id;

/**
 * Timestamp rounded to current day by dividing by 86400.
 * @type {!number}
 * @export
 */
API.Client.TokenV3DayDataDTO.prototype.date;

/**
 * Pointer to token.
 * @type {!string}
 * @export
 */
API.Client.TokenV3DayDataDTO.prototype.token;

/**
 * Volume in token units.
 * @type {!string}
 * @export
 */
API.Client.TokenV3DayDataDTO.prototype.volume;

/**
 * Volume in derived USD.
 * @type {!string}
 * @export
 */
API.Client.TokenV3DayDataDTO.prototype.volumeUsd;

/**
 * Volume in USD even on pools with less reliable USD values.
 * @type {!string}
 * @export
 */
API.Client.TokenV3DayDataDTO.prototype.untrackedVolumeUsd;

/**
 * Liquidity across all pools in token units.
 * @type {!string}
 * @export
 */
API.Client.TokenV3DayDataDTO.prototype.totalValueLocked;

/**
 * Liquidity across all pools in derived USD.
 * @type {!string}
 * @export
 */
API.Client.TokenV3DayDataDTO.prototype.totalValueLockedUsd;

/**
 * Price at end of period in USD.
 * @type {!string}
 * @export
 */
API.Client.TokenV3DayDataDTO.prototype.priceUsd;

/**
 * Fees in USD.
 * @type {!string}
 * @export
 */
API.Client.TokenV3DayDataDTO.prototype.feesUsd;

/**
 * Opening price USD.
 * @type {!string}
 * @export
 */
API.Client.TokenV3DayDataDTO.prototype.open;

/**
 * High price USD.
 * @type {!string}
 * @export
 */
API.Client.TokenV3DayDataDTO.prototype.high;

/**
 * Low price USD.
 * @type {!string}
 * @export
 */
API.Client.TokenV3DayDataDTO.prototype.low;

/**
 * Close price USD.
 * @type {!string}
 * @export
 */
API.Client.TokenV3DayDataDTO.prototype.close;

